File: recursion/s9-problems-2/exercise-19/solution.js
/*
Write a recursive function called capitalizeFirst.
Given an array of strings, capitalize the first letter of each string in the array.
*/

function capitalizeFirst(arr) {
  let word = arr[0];
  word = word.charAt(0).toUpperCase() + word.slice(1);
  if (arr.length === 1) return [word];

  return [word, ...capitalizeFirst(arr.slice(1))];
}

console.log(capitalizeFirst(["car", "taco", "banana"]));

File: recursion/s9-problems-2/exercise-19/solution.py
def capitalize_first(arr):
    word = arr[0]
    word = word[0].capitalize() + word[1:]

    if len(arr) == 1:
        return [word]

    return [word] + capitalize_first(arr[1:])


print(capitalize_first(["car", "taco", "banana"]))
